ts/components/conversation/message/message-content/MessageBody.tsx

```diff
@@ -20,21 +20,28 @@ type Props = {
   /** If set, links will be left alone instead of turned into clickable `<a>` tags. Used in quotes, convo list item, etc */
   disableLinks: boolean;
   isGroup: boolean;
+  isPublic: boolean;
 };
 
-const renderMentions: RenderTextCallbackType = ({ text, key, isGroup }) => (
-  <AddMentions key={key} text={text} isGroup={isGroup} />
+const renderMentions: RenderTextCallbackType = ({ text, key, isGroup, isPublic }) => (
+  <AddMentions key={key} text={text} isGroup={isGroup} isPublic={isPublic} />
 );
 
 export const renderTextDefault: RenderTextCallbackType = ({ text }) => <>{text}</>;
 
-const renderNewLines: RenderTextCallbackType = ({ text: textWithNewLines, key, isGroup }) => {
+const renderNewLines: RenderTextCallbackType = ({
+  text: textWithNewLines,
+  key,
+  isGroup,
+  isPublic,
+}) => {
   return (
     <AddNewLines
       key={key}
       text={textWithNewLines}
       renderNonNewLine={renderMentions}
       isGroup={isGroup}
+      isPublic={isPublic}
```

Comment:
not too sure if just calling useSelectedIsPublic would have not been nice that this prop drilling , but it also depends how this is being used. For instance if that's rendered in the left pane `useSelectedX` wouldn't work